### PR TITLE
Added optional _px variable in flavours for ghost labelled jets

### PR DESCRIPTION
**Pull request by WeiShengL**
## Summary

This pull request introduces the following changes
* Added optional `_px` variable for flavours so that `ghostbjets` and `bjets` would map to the same probability output `pb`

## Conformity
- [ ] [Changelog entry](https://github.com/umami-hep/atlas-ftag-tools/blob/main/changelog.md)
- [ ] [Documentation](https://umami-hep.github.io/atlas-ftag-tools/)

### CODE REVIEW

**Comment by samvanstroud** [comment]
Thanks for this @WeiShengL! I think it probably makes sense to merge this for now, but in the longer term I think we should require that different labels can map to the same string for probability outputs (e.g. `bjets` and `ghostbjets` would both be `pb`).

I can think of two ways to do this:
1. Define different categories as nested lists rather than flat lists, so that the label is fully specified by `category.label_name` rather than just `label_name`. this would mean `label_name` no longer needs to be unique
2. Add a new property for the `px` explicitly, which could default to the current one if not specified. This is probably the simpler option for now.

What do you think about this approach? To me it feels a bit more sustainable than adding a bunch of discriminant functions

**Comment by WeiShengL** [comment]
Yea I agree it does not look good with the different discriminant functions.

For approach 1, do you mean having something like this in the flavours.yaml?
```
single-btag:
  - name: bjets
    label: $b$-jets
    cuts: ["HadronConeExclTruthLabelID == 5"]
    colour: tab:blue

single-btag-ghost:
  - name: bjets
    label: $b$-jets
    cuts: ["HadronGhostTruthLabelID == 5"]
    colour: tab:blue
  ```

**Comment by samvanstroud** [comment]
So we already have the `label`, I meant also to have a `px` or similar key, for e.g.
```yaml
single-btag:
  - name: bjets
    label: $b$-jets
    px: b
    cuts: ["HadronConeExclTruthLabelID == 5"]
    colour: tab:blue

single-btag-ghost:
  - name: bjets
    label: $b$-jets
    px: b
    cuts: ["HadronGhostTruthLabelID == 5"]
    colour: tab:blue
```
It might be redundant to have both the `label` and the `px`, we could probably build the `label` from the `px`.

File: ftag/flavour.py [hunk 1]
```diff
@@ -22,14 +22,11 @@ class Flavour:
     cuts: Cuts
     colour: str
     category: str
-    _px: str = None
+    _px: str = ""
```
Comment (samvanstroud) [inline ftag/flavour.py:25]: I think better to use `None` as the default, it's a bit safer

File: ftag/wps/discriminant.py [hunk 2]
```diff
@@ -112,6 +112,7 @@ def get_discriminant(
         "taujets": tautag_dicriminant,
         "hbb": hbb_discriminant,
         "hcc": hcc_discriminant,
+        "ghostbjets": btag_discriminant,
```
Comment (samvanstroud) [inline ftag/wps/discriminant.py:115]: I guess we can remove this?

File: ftag/tests/wps/test_discriminant.py [hunk 3]
```diff
@@ -202,6 +202,21 @@ def test_get_discriminant():
...
+        ],
+        dtype=[("tagger_pb", "f4"), ("tagger_pc", "f4"), ("tagger_pu", "f4")],
+    )
+
+    tagger = "tagger"
+    signal = Flavours.ghostbjets
+    disc = get_discriminant(jets, tagger, signal, fc=0.1)
+    expected = btag_discriminant(jets, tagger, fc=0.1)
+    assert np.allclose(disc, expected)
+
```
Comment (samvanstroud) [inline ftag/tests/wps/test_discriminant.py:219]: is this doing anythig different to the above lines now?

File: ftag/flavour.py (hunk 1 above)
Comment (WeiShengL) [reply ftag/flavour.py:25]: I remember getting some warning from pre-commit when I was using `None` but let me try again

File: ftag/tests/wps/test_discriminant.py (hunk 3 above)
Comment (WeiShengL) [reply ftag/tests/wps/test_discriminant.py:219]: Not really I was thinking to leave it there to make sure that ghost jets are set up to the right px but I could probably do with a more simple test

**Comment by WeiShengL** [comment]
We should probably be careful when we would like to merge this in so that people don't end up with different models with pb or pghostb. Unless it doesn't really matter?

**Comment by samvanstroud** [comment]
> We should probably be careful when we would like to merge this in so that people don't end up with different models with pb or pghostb. Unless it doesn't really matter?

Yeah good point, maybe worth merging the ghost discriminant for now so people can make plots with the existing trainings... then we can open an issue to remember to remove it later :)

File: ftag/flavour.py (hunk 1 above)
Comment (samvanstroud) [reply ftag/flavour.py:25]: yeah you need to update the type hint :)

**Comment by WeiShengL** [comment]
I'll mark this as draft at the moment and created a PR #99 for the temporary ghost discriminant. Maybe we'll merge this in before preprocessing the samples for high stats training?

**Comment by samvanstroud** [comment]
Thanks! I'm happy to merge this now, any reason to keep it as draft?

**Comment by WeiShengL** [comment]
I think this does not go well with #99. Even if I have 

```
get_discriminant(jets, tagger, signal=Flavour.ghostbjets)
```
It will look for `pb` [here](https://github.com/umami-hep/atlas-ftag-tools/blob/9b03f812609eb74ff4a35a37d46d59dde72fd538/ftag/wps/discriminant.py#L53)

**Comment by samvanstroud** [comment]
I see thanks, for now can we add a fallback if `pb` is not found to look for a string like this instead? https://github.com/umami-hep/atlas-ftag-tools/blob/main/ftag/flavour.py#L28

**Comment by samvanstroud** [comment]
Thanks Wei
